andela/travel_tool_front
# #161646967 Fix yarn version inconsistency

#### What does this PR do?
 - Introduces integrity hashes to yarn.lock dependencies to avoid inconsistencies
 - Introduces a check for the yarn version being used to perform yarn commands
 - Modifies Dockerfile and circle CI to install version 1.12.x of yarn.

#### Description of Task to be completed?
- Use the last known working [yarn.lock](https://raw.githubusercontent.com/andela/travel_tool_front/1614a7582fa5e1dd0875b72404ff0eb22411b61a/yarn.lock) file to generate integrity hashes for each dependency by simply running `yarn install` with the said yarn.lock file in the root of the working directory.
- Introduce a check for the yarn version being used to perform yarn commands
- Modifiy Dockerfile and circle CI to install version 1.12.x of yarn.

#### How should this be manually tested?
 - Clone the repo
    `git clone https://github.com/andela/travel_tool_front.git`
 - Switch to the project directory
   `cd travel_tool_front`
 - Checkout the **fix-yarn-lock-inconsistencies-161775747** branch
   `git checkout fix-yarn-lock-inconsistencies-161775747`
 - Using `yarn`:
    - Run `yarn install` to install dependencies.
       It is expected that the yarn.lock file will not be modified and that the output on the terminal will look close to this:
![image](https://user-images.githubusercontent.com/41262970/48264568-fd144400-e43a-11e8-8359-a76237fdc92c.png)

       Please note that `Saved lockfile` is not printed on the terminal.
       Please note that the yarn command is run with `yarn v1.12.3`.

     - Run `yarn check --integrity`
It is expected that the check succeeds and that the output in the terminal will look close to this:
![image](https://user-images.githubusercontent.com/41262970/48264674-61cf9e80-e43b-11e8-9d04-54523a9e0663.png)

     - Run `yarn start`
     - Start the [backend](https://github.com/travel_tool_back) of this project.
     - Navigate to [travela](http://travela-local.andela.com:3000/) in your browser and log in with your Andela account. 
     - Try and create a new request. It is expected that you are able to type on the input fields. If this is not the case, please mention it in the comments under this PR. **This is an important step.**

- Using `make`:
    - Run `make start`
       It is expected that the steps 1 through 9 run without fail, and that a new lockfile is not created in the docker environment. The output should look like this as from step 7:
![image](https://user-images.githubusercontent.com/41262970/48264374-562fa800-e43a-11e8-9c98-ad09c1656b82.png)

       Please note that `Saved lockfile` is not printed to the terminal.
       Please note that the yarn command is run with `yarn v1.12.3`.

    - Run `make ssh`.
       This will let you access the terminal of the docker environment where your app is running.
       - Inside this environment, run `yarn check --integrity`
          It is expected that the check succeeds and that the output in the terminal will look close to this:
![image](https://user-images.githubusercontent.com/41262970/48264221-d0abf800-e439-11e8-8c2b-80b12342ca87.png)

#### Any background context you want to provide?
* The **Using `make`** part of testing this PR needs one to have already installed docker.
* You will need yarn version 1.12.x to perform the yarn tests mentioned above, otherwise an error is raised as shown below:
  ![image](https://user-images.githubusercontent.com/41262970/48263383-fbe11800-e436-11e8-9ca5-56a7ea038a19.png)

#### What are the relevant pivotal tracker stories?
- [#161646967](https://www.pivotaltracker.com/story/show/161775747)

#### Questions:
N/A
